zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnStateTransitionBehavior.java

```diff
@@ -461,42 +462,50 @@ public <T extends ExecutableFlowNode> void takeOutgoingSequenceFlows(
    * @param element the flow node element that is being processed
    * @param context the context of the process instance
    */
-  public <T extends ExecutableFlowNode> void executeRuntimeInstructionsIfNeeded(
-      final T element, final BpmnElementContext context) {
+  public <T extends ExecutableFlowNode>
+      Either<ProcessInstanceInterrupted, BpmnElementContext> executeRuntimeInstructions(
+          final T element, final BpmnElementContext context) {
```

Comment:
🔧 This method's Javadoc should make it clear what the left and right returned values are and when it returns which